services/store.go

```diff
@@ -181,71 +186,46 @@ outer:
 					if !ok {
 						continue
 					}
-					xfer := decodeTransfer(height, info, evt)
-					if xfer == nil {
-						log.Warnf(
-							"No transfer detected for state %#v in transaction %s at height %d",
-							evt.States, info.TxHash.ToHexString(), height,
-						)
-						continue
-					}
-					if failed {
-						if evt.ContractAddress != ongAddr {
+					//check evm ong event log
+					isEvm, eventLog := checkEvmEventLog(evt)
+					if isEvm {
+						xfer, err := parseEvmOngTransferLog(eventLog)
+						if err != nil {
+							log.Warnf("prase evm ong err:%s,height:%d,txhash:%s", err, info.TxHash.ToHexString(), height)
 							continue
 						}
-						// NOTE(tav): We skip additional transfer events on
-						// failed transactions if we've already matched the gas
-						// fee.
-						if gasVerified {
+						balanceCal(xfer, evt, diffs, txn.Transfers)
+					} else {
+						xfer := decodeTransfer(height, info, evt)
+						if xfer == nil {
+							log.Warnf(
+								"No transfer detected for state %#v in transaction %s at height %d",
+								evt.States, info.TxHash.ToHexString(), height,
+							)
 							continue
 						}
-					}
-					if gasVerified {
-						xfer.isGas = false
-					} else if xfer.isGas {
-						if ori.Payer == xfer.from {
-							gasVerified = true
-						} else {
-							xfer.isGas = false
-						}
-					}
-					if xfer.from != nullAddr {
-						accts, ok := diffs[xfer.from]
-						if ok {
-							balance, ok := accts[evt.ContractAddress]
-							if ok {
-								accts[evt.ContractAddress] = balance.Sub(balance, xfer.amount)
-							} else {
-								accts[evt.ContractAddress] = (&big.Int{}).Neg(xfer.amount)
+						if failed {
```

Comment:
Should this section of logic also not be shared for the `isEvm` block?

Reply:
I don't think it is necessary to come out if the EVM transaction fails. Just accept an Eventlog about the consumption of ONG as a handling fee, what do you think?

Reply:
yes,I have added, in the latest code